wireapp/kalium
# feat: add mega linter for code quality check

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

- Add extra security/quality checks and auto-fixes for code formatting to the CI

### Solutions

- Added [mega-linter](https://github.com/oxsecurity/megalinter#megalinter-by-) checks to the CI on every push and pull request 


#### How to Test

As soon as the workflow will be accepted by a maintainer and workflow `run-code-quality-check` will complete you should see mega linter status (example in attachments). 

To make auto-fixes functionality work someone needs to add his Personal Access token to the repository secrets or environment. As described[ here ](https://github.com/oxsecurity/megalinter#apply-fixes-issues). With method 2 no changes in code would be needed. 

### Notes

There are plenty of possible configurations: 
- removing not wanted linters
- adding configuration files for specific linters
- choosing to validate all code base or not (right now it is set to launch only on push to main)
- choosing how to apply auto-fixes - by auto-commit or by creating additional PR with changes. 
See [mega-linter](https://github.com/oxsecurity/megalinter#megalinter-by-) for more. 


### Attachments (Optional)

Example check: 
![Screenshot 2023-05-29 at 14 03 31](https://github.com/wireapp/kalium/assets/33498031/98516a90-a07d-428f-a44e-2615157b9b09)

Finding specific issue:

![Screenshot 2023-05-29 at 14 09 20](https://github.com/wireapp/kalium/assets/33498031/202c7d7a-02ab-44c9-a15b-5c1f4fa4ac69)

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
